frame/support/src/traits.rs

```diff
@@ -1518,6 +1518,36 @@ pub trait OnFinalize<BlockNumber> {
 	fn on_finalize(_n: BlockNumber) {}
 }
 
+/// The block's on idle trait
+///
+/// Implementing this lets you express what should happen for your pallet
+/// before on finalize and based on if a remaining weight threshold is met.
```

Comment:
```suggestion
/// The block's on idle trait.
///
/// Implementing this lets you express what should happen for your pallet
/// before finalization in case there is remaining weight left for the block.
```